src/YesSql.Provider.MySql/MySqlDbProviderOptionsExtensions.cs

```diff
@@ -31,8 +42,18 @@ public static class MySqlDbProviderOptionsExtensions
             }
 
             configuration.ConnectionFactory = new DbConnectionFactory<MySqlConnection>(connectionString);
-            configuration.DocumentStorageFactory = new SqlDocumentStorageFactory();
             configuration.IsolationLevel = isolationLevel;
+
+            if (cached)
+            {
+                configuration.DocumentStorageFactory = 
+                    new CacheDocumentStorageFactory(new SqlDocumentStorageFactory());
+            }
+            else
+            {
+                configuration.DocumentStorageFactory = 
+                    new CacheDocumentStorageFactory(new SqlDocumentStorageFactory());
```

Comment:
It's not a duplicate, almost the providers here are using `new CacheDocumentStorageFactory(new SqlDocumentStorageFactory())` so that's why it appears as duplicate 😄